runtimes/kotlin-1.6/src/main/kotlin/io/openruntimes/kotlin/Server.kt

```diff
@@ -8,56 +8,173 @@ import java.io.StringWriter
 import java.io.PrintStream
 import java.io.PrintWriter
 import kotlinx.coroutines.runBlocking
+import java.net.URLEncoder
+import java.nio.charset.StandardCharsets
+import java.io.UnsupportedEncodingException
+import com.google.gson.GsonBuilder
+import com.google.gson.Gson
+import java.lang.reflect.Method
+import java.util.HashMap
 
 suspend fun main() {
     Javalin
         .create()
         .start(3000)
-        .post("/") { runBlocking { execute(it) } }
+        .get("/*") { runBlocking { execute(it) } }
+        .post("/*") { runBlocking { execute(it) } }
+        .put("/*") { runBlocking { execute(it) } }
+        .delete("/*") { runBlocking { execute(it) } }
+        .patch("/*") { runBlocking { execute(it) } }
+        .options("/*") { runBlocking { execute(it) } }
+        .head("/*") { runBlocking { execute(it) } }
 }
 
 suspend fun execute(ctx: Context) {
-    if (ctx.header("x-internal-challenge").isNullOrBlank()) {
-        ctx.status(500).result("Unauthorized");
-        return;
+    var secret = ctx.header("x-open-runtimes-secret");
+    var serverSecret = System.getenv("OPEN_RUNTIMES_SECRET");
+
+    if(secret == null) {
+        secret = "";
+    }
+
+    if(serverSecret == null) {
+        serverSecret = "";
     }
-    if (ctx.header("x-internal-challenge") != System.getenv("INTERNAL_RUNTIME_KEY")) {
-        ctx.status(500).result("Unauthorized")
+
+    if (secret.equals("") || secret != serverSecret) {
+        ctx.status(500).result("Unauthorized. Provide correct \"x-open-runtimes-secret\" header.")
         return
     }
 
-    val codeWrapper = Wrapper()
-    val request = RuntimeRequest(ctx)
-    val response = RuntimeResponse()
+    var bodyString: String = ctx.body()
+    var body: Any = bodyString
+    var headers: MutableMap<String, String> = mutableMapOf<String, String>()
+    var method: String = ctx.method()
+
+    for (entry in ctx.headerMap().entries.iterator()) {
+        var header: String = entry.key.lowercase()
+        if(!(header.startsWith("x-open-runtimes-"))) {
+            headers.put(header, entry.value)
+        }
+    }
+
+    var contentType: String = ctx.header("content-type") ?: "text/plain"
+    if(contentType.contains("application/json")) {
+        if(!bodyString.isEmpty()) {
+            var gson = GsonBuilder().serializeNulls().create();
+            body = gson.fromJson(bodyString, MutableMap::class.java)
+        } else {
+            body = HashMap<String, Any>();
+        }
+    }
+
+    var hostHeader = ctx.header("host");
+    var protoHeader = ctx.header("x-forwarded-proto");
+
+    if(hostHeader == null) {
+        hostHeader = "";
+    }
+
+    if(protoHeader == null) {
+        protoHeader = "http";
+    }
+
+    var scheme: String = protoHeader
+    var defaultPort: String = if (scheme.equals("https")) "443" else "80"
```

Comment:
This can be an `Int`

Reply:
I need to apply this in all runtimes. Keeping it like this here, and making myself note to update in all runtimes in a separate PR.